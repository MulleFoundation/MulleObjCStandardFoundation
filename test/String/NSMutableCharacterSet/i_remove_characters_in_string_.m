#ifdef __MULLE_OBJC__
# import <MulleObjCStandardFoundation/MulleObjCStandardFoundation.h>
# include <mulle-testallocator/mulle-testallocator.h>
#else
# import <Foundation/Foundation.h>
#endif
#include <stdio.h>
#include <stdlib.h>
#if defined(__unix__) || defined(__unix) || (defined(__APPLE__) && defined(__MACH__))
# include <unistd.h>
#endif


static int   test_i_remove_characters_in_string_( void)
{
   NSMutableCharacterSet *obj;
   NSString * params_1[] =
   {
      [NSMutableString stringWithString:@"@1848"],
      [NSMutableString stringWithString:@"VfL Bochum 1848"],
      [NSMutableString stringWithString:@"Lorem ipsum dolor sit amet, consectetur adipisici elit, sed eiusmod tempor incidunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquid ex ea commodi consequat. Quis aute iure reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur."] /* > 256 chars */,
      [[[NSString alloc] initWithBytes:"\xe2\x98\x84\xef\xb8\x8f\xe2\x98\x83\xef\xb8\x8f\xf0\x9f\x91\x8d\xf0\x9f\x8f\xbe" /* emoji comet, snowman, thumbs-up brown */ length:20 encoding:NSUTF8StringEncoding] autorelease],
      [NSMutableString stringWithString:@""],
      nil
   };
   unsigned int   i_1;
   unsigned int   i_2;
   unsigned int   n_1 = sizeof( params_1) / sizeof( NSString *);

   for( i_1 = 0; i_1 < n_1; i_1++)
   {
      for( i_2 = 0; i_2 < n_1; i_2++)
      {
         @try
         {
            obj = [NSMutableCharacterSet characterSetWithCharactersInString:params_1[ i_2]];
            [obj removeCharactersInString:params_1[ i_1]];
            printf( "%u/%u: %s\n", i_1, i_2, [[obj mulleTestDescription] UTF8String]);
         }
         @catch( NSException *localException)
         {
            printf( "Threw a %s exception\n", [[localException name] UTF8String]);
         }
      }
   }
   return( 0);
}


int   main( int argc, char *argv[])
{
   int   rval;

   rval = test_i_remove_characters_in_string_();
   return( rval);
}

