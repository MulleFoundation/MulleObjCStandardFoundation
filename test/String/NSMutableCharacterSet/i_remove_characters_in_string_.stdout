0/0: 
0/1:  
B
L
V
c
f
h
m
o
u
0/2:  
,
.
L
Q
U
a
b
c
d
e
f
g
h
i
l
m
n
o
p
q
r
s
t
u
v
x
0/3: ☃
☄
️
🏾
👍
0/4: 
0/5: 
1/0: @
1/1: 
1/2: ,
.
Q
U
a
b
d
e
g
i
l
n
p
q
r
s
t
v
x
1/3: ☃
☄
️
🏾
👍
1/4: 
1/5: 
2/0: 1
4
8
@
2/1: 1
4
8
B
V
2/2: 
2/3: ☃
☄
️
🏾
👍
2/4: 
2/5: 
3/0: 1
4
8
@
3/1:  
1
4
8
B
L
V
c
f
h
m
o
u
3/2:  
,
.
L
Q
U
a
b
c
d
e
f
g
h
i
l
m
n
o
p
q
r
s
t
u
v
x
3/3: 
3/4: 
3/5: 
4/0: 1
4
8
@
4/1:  
1
4
8
B
L
V
c
f
h
m
o
u
4/2:  
,
.
L
Q
U
a
b
c
d
e
f
g
h
i
l
m
n
o
p
q
r
s
t
u
v
x
4/3: ☃
☄
️
🏾
👍
4/4: 
4/5: 
5/0: 1
4
8
@
5/1:  
1
4
8
B
L
V
c
f
h
m
o
u
5/2:  
,
.
L
Q
U
a
b
c
d
e
f
g
h
i
l
m
n
o
p
q
r
s
t
u
v
x
5/3: ☃
☄
️
🏾
👍
5/4: 
5/5: 
